generator/src/main/java/org/corfudb/generator/distributions/DataSet.java

```diff
@@ -0,0 +1,33 @@
+package org.corfudb.generator.distributions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Created by maithem on 7/14/17.
+ */
+public abstract class DataSet {
+
+    public abstract void populate();
+
+    public List sample(int num) {
+        getDataSet().size();
+
+        List<String> list = new ArrayList<>();
+
+        if (num > getDataSet().size()) {
+            for (int x = 0; x < (num / getDataSet().size()) + 1; x++) {
+                list.addAll(getDataSet());
+            }
+        } else {
```

Comment:
seems like the 'else' is just the normal case with (num /  getDataSet().size()) equaling zero.

more generally, I don't know why do it this way -- why not use a random number generator with a deterministic seed?